File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.js
/**
 * @param {number[]} target
 * @param {number} n
 * @return {string[]}
 */
var buildArray = function(target, n) {
    let res = [];
    j = 0;
    for(let i=1;i<=n;i++) {
        if(target[j]===i) {
            res.push('Push');
            j++;
        } else res.push(...['Push','Pop']);
        if(j===target.length) return res;
    }
    return res;
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.java
public class Solution {
    public List<String> buildArray(int[] target, int n) {
        HashSet<Integer> targetSet = new HashSet<>();
        for (int num : target) {
            targetSet.add(num);
        }

        List<String> result = new ArrayList<>();
        for (int i = 1; i <= target[target.length - 1]; i++) {
            if (targetSet.contains(i)) {
                result.add("Push");
            } else {
                result.add("Push");
                result.add("Pop");
            }
        }
        return result;
    }
}